rcl_yaml_param_parser/src/namespace.c

```diff
@@ -16,6 +16,19 @@
 
 #include "rcutils/strdup.h"
 
+///
+/// Validate a name whether it is a valid namespace
+///
+static rcutils_ret_t
+__validate_namespace(const char * name)
```

Comment:
why it should not be dependent on `rmw`? could you elaborate a little bit? I think it would be better to use `rmw` to validate the namespace instead of having redundant code else where.

> I found that using rmw_validate_namespace has some limitations because there are some extra rules, such as "/**" (maybe more in the future), need to be parsed.

you mean https://github.com/ros2/rclcpp/issues/1265?

Reply:
Thanks for your questions, @fujitatomoya 

> why it should not be dependent on rmw?

"/**" is a special rule at `rclcpp`, it is invalid checked by `rmw_validate_namespace`.
If using `rmw_validate_namespace`, we need to add more extra checking for these special rules inside `__validate_namespace`, which seems not a requirement of this issue.

>> maybe more in the future

> you mean ros2/rclcpp#1265?

Yes.

Reply:
Though this solves the reported issue, I think we should do a more general check if the node namespace (and node name) are valid. I would do this using `rmw_validate_namespace` and `rmw_validate_node_name`, and handle the special cases for wildcards `*` and `**`.

I suggest implementing this function as follows:

1. Make a copy of the name to validate
2. Substitute substrings `*` and `**`  with arbitrary valid strings (e.g. `WILDCARD`)
3. Pass the modified name to `rmw_validate_namespace`

By using the rmw functions we are consistent with what it means to be a valid name.

What do you think?

Reply:
@jacobperron

> I suggest implementing this function as follows

Thanks for your suggestion. I'll update it later.